crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,823 @@
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Open Questions
+# --------------
+# - Rules for nullabilit/non-nullability
+# - Pattern for multi-gets
+# - Encoding RPC errors
+# - `type` (vs `scalar`) for Move Types
+# - Pattern for upcasting and downcasting
+
+# Check Completeness Against Existing APIs
+# -------------
+# DONE Read
+# DONE - Objects
+# DONE - Transactions
+# DONE - Checkpoints
+# DONE - Events
+# DONE - ProtocolConfig
+# DONE - Chain Identifier
+# DONE Coin
+# DONE Extended
+# DONE Governance
+# DONE Indexer
+# DONE Move Utils
+# TODO Write
+
+# DONE Port designs from Notion
+# DONE - Display
+# DONE - SuiNS
+# DONE Data Consistency API
+# TODO Flesh out Connections
+# TODO - TotalTransactionBlocks for the PTB connection
+
+schema {
+  query: Query
+  subscription: Subscription
+}
+
+type Query {
+  # First four bytes of the network's genesis checkpoint digest
+  # (uniquely defines the network)
+  chainIdentifier: String!
+
+  # First and last checkpoint that the RPC can answer consistent
+  # queries for.
+  consistentRange: ConsistentRange
+
+  owner(address: SuiAddress!): Owner
+  object(address: SuiAddress!, version: BigInt): Object
+  address(address: SuiAddress!): Address
+
+  epoch(epochId: Int): Epoch
+  protocolConfig(protocolVersion: Int): ProtocolConfig
+
+  checkpoint(filter: CheckpointFilter): Checkpoint
+  checkpoints(order: Order!): CheckpointConnection
```

Comment:
From my research on connections, it seems naming the fields more explicitly helps in case we want to deprecate a certain pagination type down the road.
So for example instead of `checkpoints(order: Order!): CheckpointConnection`, it would be `checkpointsConnection(order: Order!): CheckpointConnection`

Reply:
I'm fine doing that, it makes it obvious when you're dealing with a paginated set of items vs a flat list, although I really don't think it's _essential_